src/CMakeLists.txt

```diff
@@ -253,40 +254,57 @@ source_group("types" FILES ${types_files})
 source_group("utils" FILES ${utils_files})
 
 #--------------------------------------------------------
-# Lib
-add_library(${src_lib_static_nupiccore_solo} STATIC 
+# Lib core_alg (encoders, SP, TM/TP/BackTM, Anomaly,...) 
+add_library(${src_lib_static_nupiccore_alg} STATIC 
     ${algorithm_files} 
     ${encoders_files} 
-    ${engine_files} 
     ${math_files} 
-    ${ntypes_files} 
-    ${os_files} 
-    ${regions_files} 
     ${types_files} 
     ${utils_files} 
 )
 
 
-target_compile_options( ${src_lib_static_nupiccore_solo} PUBLIC ${INTERNAL_CXX_FLAGS})
-target_compile_definitions(${src_lib_static_nupiccore_solo} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
-target_include_directories(${src_lib_static_nupiccore_solo} PRIVATE 
+target_compile_options( ${src_lib_static_nupiccore_alg} PUBLIC ${INTERNAL_CXX_FLAGS} ${CXX_FLAGS_OPTIMIZED})
```

Comment:
separate build units src/nupic/algorithms (gets optimization flags, Ofast) 
and rest (netoworkAPI)

Reply:
Is there a reason that the optimization flags are not / should not be applied to the "rest"/networkAPI?

Reply:
> Is there a reason that the optimization flags are not / should not be applied to the "rest"/networkAPI?

the algorithms/ folder is the only workhorse here. 

> "rest"/networkAPI?

they could, but
- neglible/negative impact
- longer compile time
- i wanted to separate hard-core optimization vs rather security checks focused rest

Reply:
I disagree.  networkAPI needs optimization as much as algorithms.  networkAPI does not use as much resources as algorithms but every little bit helps.

Reply:
> I disagree. networkAPI needs optimization as much as algorithms. networkAPI does not use as much resources as algorithms but every little bit helps. 

Ok, what's true...we can't decide now, there's no functioning NetworkAPI benchmark. We should test with the NetworkAPI version of Hotgym benchmark, when all parts are available. 

I suspect compiler optimizations won't matter there much, as NAPI constructs the network, moves data around (ideally pointers) and instruments execution of the algs (optimized here). 

See what `Ofast` enables, it may help performance, but also may harm it, those features are highly task specific. 

TL;DR: we should revisit with Networkapi hotgym benchmark and choose the best options for each. (Ofast does not necessarily mean best)